charts/edge-webrtc-service/templates/deployment.yaml

```diff
@@ -0,0 +1,88 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include ".helm.fullname" . }}
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include ".helm.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include ".helm.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include ".helm.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include ".helm.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        {{- if .Values.sidecarimage }}
+        - name: "{{ .Values.sidecarimage.name }}"
+          image: "{{ .Values.sidecarimage.repository }}:{{ .Values.sidecarimage.tag }}"
+          imagePullPolicy: {{ .Values.sidecarimage.pullPolicy }}
+        {{- end }}      
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.image.env }}
+          env: 
+            {{- toYaml .Values.image.env | nindent 12 }}
+          {{- end }}
+          args: [ {{.Values.image.args}} ]
+          ports:
+            - name: http
+              containerPort: {{ .Values.image.containerPort }}
+              protocol: TCP
+          livenessProbe:
+            exec:
+              command:
+              - cat
+              - /opt/iot/certificate.pem
+            initialDelaySeconds: 5
+            periodSeconds: 5              
+          readinessProbe:
+            exec:
+              command:
+              - cat
+              - /opt/iot/certificate.pem
+            initialDelaySeconds: 5
+            periodSeconds: 5  
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+            {{- if .Values.iot.secret }}
+          volumeMounts:
+            - name: iot-secret
+              mountPath: /opt/iot
+              readOnly: true
+            {{- end }}            
+        {{- if .Values.iot.secret }}
+      volumes:
+        - name: iot-secret
+          secret:
+            secretName: {{ .Values.iot.secret }}
+        {{- end }}
```

Comment:
could probably go with a single if condition here if we don't plan on mounting any other volumes